HIP-XXXX.md

```diff
@@ -0,0 +1,298 @@
+
+```
+TBSCertificate  ::=  SEQUENCE  {
+        version         [0]  EXPLICIT Version DEFAULT v1,
+        serialNumber         CertificateSerialNumber,
+        signature            AlgorithmIdentifier,
+        issuer               Name,
+        validity             Validity,
+        subject              Name,
+        subjectPublicKeyInfo SubjectPublicKeyInfo,
+        issuerUniqueID  [1]  IMPLICIT UniqueIdentifier OPTIONAL,
+                             -- If present, version MUST be v2 or v3
+        subjectUniqueID [2]  IMPLICIT UniqueIdentifier OPTIONAL,
+                             -- If present, version MUST be v2 or v3
+	extensions      [3]  EXPLICIT Extensions OPTIONAL
+                             -- If present, version MUST be v3		
+```
+
+In the extensions field, we can define custom extensions. Each extension has an OID (Object identifier, a unique id) called `extnID` and the value of the extension `extnValue` is an ASN.1 DER encoded structure. An ASN.1 OCTET STRING should be suitable for storing arbitrary data.
+
+
+
+### 1.0 Urkel proof extension:
+
+```
+extnId: TBD
+extnValue: n<urkel root><urkel proof>,...
+```
+
+The following sections describe what to include in the `extnValue` field depending on UPDATEs to the name within the last 32 urkel intervals.
+
+### 1.1 No UPDATEs:
+
+Certificates MUST be generated weekly and should include the 32nd most recent tree root and proof. This is the most common case.
+
+### 1.2 NS modified:
+
+Exactly the same as 1.1. Servers MUST NOT include the new NS records. If its a HIP-5 delegation, refer to 1.3.
+
+### 1.3 HIP-5 NS modified:
+
+HIP-5 delegations are special NS records. They typically combine the security properties of NS+DS. During this transition period, servers MUST include two proofs in the certificate for the current and previous NS records.
+
+- Proof 1: The 32nd most recent urkel root/proof from the previous UPDATEs.
+- Proof 2: The urkel root/proof at the time the new UPDATE was committed.
+
+### 1.4 DS rollover:
+
+This case requires special care to accomodate clients with older tree roots. During this transition period, servers MUST include two proofs in the certificate for the current and previous DS records. Servers MUST sign the DNS RRSets with both DS records and include it in the embedded DNSSEC chain extension.
+
+- Proof 1: The 32nd most recent urkel root/proof from the previous UPDATEs.
+- Proof 2: The urkel root/proof at the time the new UPDATE was committed.
+
+
+### 2.0 DNSSEC Authentication chain
+
+```
+extnID: TBD
+extnValue: MUST use the same serialization format as described in RFC9102.
+```
+
+If the urkel extension includes DS records, include the complete DNSSEC authentication chain.
+
+
+
+## ACME clients & servers
+
+Servers MAY use ACME services (RFC8555) for certificate issuance that embed the appropriate x509 certificate extensions described above using a standard ACME client such as certbot without having to install an HNS client.

```

Comment:
An ACME client can also choose to run hnsd to verify its own proofs, provided by the ACME service